modules/json_form_widget/README.md

```diff
@@ -0,0 +1,557 @@
+
+
+## Installation
+
+The forms that the JSON Form Widget creates utilize the DKAN metastore for schema discovery and subsequently create nodes as the "data" content type. This module therefore is currently packaged with DKAN and as of now cannot be installed independantly.
+
+## Configuration
+
+DKAN includes schema files that follow the [Project Open Data DCAT-US schema standards](https://resources.data.gov/resources/dcat-us/). If you need to add additional metadata fields, or wish to remove fields, simply create a new schema/collection/ directory in the root of your site (usually docroot), and add your new schema files to it.
+
+---
+
+> **_NOTE:_**
+> Read the documentation on [Changing your dataset schema](https://dkan.readthedocs.io/en/latest/user-guide/guide_custom_schemas.html) to learn how to add custom fields.
+
+---
+
+## Field Type UI Examples
+
+If we look closely at the provided DKAN [dataset schema](https://github.com/GetDKAN/dkan/blob/2.x/schema/collections/dataset.json), and [dataset UI schema](https://github.com/GetDKAN/dkan/blob/2.x/schema/collections/dataset.ui.json), we can see some examples of how different field types are converted into form elements utilizing both the dataset.json (schema file) as well as the dataset.ui.json (schema UI file) which in turn create the form @ /node/add/data?schema=dataset.
+
+---
+
+> **_TIP:_**
+> See the Schema UI Handler class as well as the Element folder and it's included class files for how different UI options are managed in code.
+
+---
+
+The following are some examples of Field types and associated options and how they appear as a form element respectively. Newly introduced schema UI options will be described only the first time they appear. The length of some JSON objects may be trimmed as compared to the provided dataset.json/ui.json files in order to keep this page's length to a minimum, but the functionality they are showcasing will remain unchanged.
+
+---
+> **_TIP:_**
+> You can make fields required on the form using a JSON property array in your schema file similarly to how it is done in the example from the dataset.json file below:
+
+---
+
+    "required": [
+      "title",
+      "description",
+      "identifier",
+      "accessLevel",
+      "modified",
+      "keyword"
+    ],
+
+In the above example the listed fields (which would appear later as objects in the JSON) would be required fields in the Drupal form that is created.
+
+### Text Field
+
+**Schema File Example:**
+
+
+    "title": {
+      "title": "Title",
+      "description": "Human-readable name of the asset. Should be in plain English and include sufficient detail to facilitate search and discovery.",
+      "type": "string",
+      "minLength": 1
+    },
+
+**UI Schema File Example:**
+
+    "title": {
+      "ui:options": {
+        "description": "Name of the asset, in plain language. Include sufficient detail to facilitate search and discovery."
+      }
+    },
+
+UI Options:
```

Comment:
Something missing here I think - the ability to add a "placeholder" to text fields and several other types of fields. For instance https://github.com/GetDKAN/dkan/blob/2.x/schema/collections/dataset.ui.json#L195

Reply:
Good eye, ty- I'll add that as a separate field example (References/url)